test/new-for-builtins.js

```diff
@@ -18,6 +26,41 @@ const disallowNewError = builtin => ({
 	message: `Use \`${builtin}()\` instead of \`new ${builtin}()\`.`
 });
 
+const enforceNew = [
+	'Object',
+	'Array',
+	'ArrayBuffer',
+	'BigInt64Array',
+	'BigUint64Array',
+	'DataView',
+	'Date',
+	'Error',
+	'Float32Array',
+	'Float64Array',
+	'Function',
+	'Int8Array',
+	'Int16Array',
+	'Int32Array',
+	'Map',
+	'WeakMap',
+	'Set',
+	'WeakSet',
+	'Promise',
+	'RegExp',
+	'Uint8Array',
+	'Uint16Array',
+	'Uint32Array',
+	'Uint8ClampedArray'
+];
```

Comment:
These are just redefinitions of what's in the rule. I think it would be better to move these to util.js and import them both in the rule and here.